quinn-proto/src/packet.rs

```diff
@@ -25,10 +25,21 @@ pub struct PartialDecode {
 }
 
 impl PartialDecode {
-    pub fn new(bytes: BytesMut, local_cid_len: usize) -> Result<Self, PacketDecodeError> {
+    pub fn new(
+        bytes: BytesMut,
+        local_cid_len: usize,
+    ) -> Result<(Self, Option<BytesMut>), PacketDecodeError> {
         let mut buf = io::Cursor::new(bytes);
         let plain_header = PlainHeader::decode(&mut buf, local_cid_len)?;
-        Ok(Self { plain_header, buf })
+        let rest = plain_header.payload_len().and_then(|len| {
+            let packet_len = (buf.position() + len) as usize;
+            if buf.get_ref().len() > packet_len {
+                Some(buf.get_mut().split_off(packet_len))
+            } else {
+                None
```

Comment:
I think we should return an error if `packet_len > buf.get_ref().len()`.

Reply:
Improved it -- though not quite the way you suggested it.